src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3783,131 +3783,164 @@ static void apply_debugger_ergo() {
   }
 }
 
-// Parse entry point called from JNI_CreateJavaVM
+static const char* const hotspotrc = ".hotspotrc";
 
-jint Arguments::parse(const JavaVMInitArgs* initial_cmd_args) {
-  assert(verify_special_jvm_flags(false), "deprecated and obsolete flag table inconsistent");
-  JVMFlag::check_all_flag_declarations();
+class Arguments::PreprocessedImpl final : public CHeapObj<mtArguments> {
```

Comment:
I preferred this named `PrepProcessedArguments`. I don't see why you needed to change this name to get rid of the void* field ???

Reply:
I can go with `PreprocessedArguments`. kimbarrett@ suggested getting rid of void* to avoid casting. I can just predeclare as PreprocessedArguments.

Reply:
Done. Replaced PreprocessedImpl with PreprocessedArguments and forward declared to get rid of void*.